test.py

```diff
@@ -135,20 +135,24 @@ def test_destroy_app(self):
 
     def test_update_frames_inclusive(self):
         """Tests for the case where a new frame is added in the return of frames()."""
-        orgFramesSet = {wrapper.widget() for wrapper in self.qiwis._wrapperWidgets["app1"]}
+        orgFramesSet = {wrapperWidget.widget()
+                        for wrapperWidget in self.qiwis._wrapperWidgets["app1"]}
         newFramesSet = orgFramesSet | {QWidget()}
-        self.qiwis._apps["app1"].frames.return_value = tuple(newFramesSet)
+        self.qiwis._apps["app1"].frames.return_value = (("title", frame) for frame in newFramesSet)
```

Comment:
It is often a better practice to _fix_ the generator values unless it requires large memory.
For example if `newFramesSet` is modified after defining this generator, and then we consume its values, the values are generated based on the modified `newFramesSet`.
Since this is a test code I think using `tuple(...)` to fix the values would be better!

Reply:
An interesting demonstration: https://stackoverflow.com/questions/70342301/closure-semantics-in-python-generator-comprehensions

Reply:
Wow, your example is too surprising! Then, I will use `tuple(...)` directly.

Reply:
Umm.. is it sufficient that `(tuple("title", frame) for frame in newFramesSet)`?

Reply:
You should write `tuple` outside: `tuple(("title", frame) for ...)`.

Reply:
I see! 👍